src/lib/wallet/GoodWalletClass.js

```diff
@@ -242,12 +242,33 @@ export class GoodWallet {
     )
   }
 
-  async pollSendEvents() {
-    const fromBlock = this.lastEventsBlock
+  async syncTxWithBlockchain(fromBlock) {
+    const toBlock = await this.wallet.eth.getBlockNumber()
+    const filter = [fromBlock, toBlock]
+
+    log.debug('Start sync of txs from blockchain', {
+      filter,
+    })
+
+    try {
+      await Promise.all([
+        this.pollSendEvents(...filter),
+        this.pollReceiveEvents(...filter),
+        this.pollOTPLEvents(...filter),
+      ])
+
+      log.debug('Sync of tx from blockchain finished successfully')
+    } catch (e) {
+      log.error('Failed to sync tx from blockchain', e.message, e)
+    }
+  }
+
+  async pollSendEvents(fromBlock = this.lastEventsBlock, toBlock) {
```

Comment:
1. optional argument should be the last one
2. default value should be some constant. using some instance var as default param value is a bad design. let set default to null and will use this.lastEventsBlock if fromBlock wasn't passed:

```javascript
async pollSendEvents(toBlock, from = null) {
  const fromBlock = from || this.lastEventsBlock
  // ...
```

Reply:
@serdiukov-o-nordwhale done